libraries/api-extractor-model/src/mixins/ApiExtendsMixin.ts

```diff
@@ -0,0 +1,316 @@
+  extendsTokenRanges: IExcerptTokenRange[];
+}
+
+export interface IApiExtendsMixinJson extends IApiItemJson {
+  extendsTokenRanges: IExcerptTokenRange[];
+}
+
+const _extendsTypes: unique symbol = Symbol('ApiExtendsMixin._extendsTypes');
+
+/**
+ * The mixin base class for API items that can extend other API items.
+ *
+ * @remarks
+ *
+ * This is part of the {@link ApiModel} hierarchy of classes, which are serializable representations of
+ * API declarations.  The non-abstract classes (e.g. `ApiClass`, `ApiEnum`, `ApiInterface`, etc.) use
+ * TypeScript "mixin" functions (e.g. `ApiDeclaredItem`, `ApiItemContainerMixin`, etc.) to add various
+ * features that cannot be represented as a normal inheritance chain (since TypeScript does not allow a child class
+ * to extend more than one base class).  The "mixin" is a TypeScript merged declaration with three components:
+ * the function that generates a subclass, an interface that describes the members of the subclass, and
+ * a namespace containing static members of the class.
+ *
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface ApiExtendsMixin extends ApiItem {
+  /**
+   * The types referenced via an "extends" heritage clause.
+   */
+  readonly extendsTypes: HeritageType[];
+
+  /**
+   * Resolves all of the ApiItem's immediate and inherited members by walking down the
+   * inheritance tree.
+   */
+  resolveInheritedMembers(): IResolveInheritedMembersResult;
+
+  /** @override */
+  serializeInto(jsonObject: Partial<IApiExtendsMixinJson>): void;
+}
+
+/**
+ * Mixin function for {@link (ApiExtendsMixin:interface)}.
+ *
+ * @param baseClass - The base class to be extended
+ * @returns A child class that extends baseClass, adding the {@link (ApiExtendsMixin:interface)} functionality.
+ *
+ * @public
+ */
+export function ApiExtendsMixin<TBaseClass extends IApiItemConstructor>(
+  baseClass: TBaseClass
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): TBaseClass & (new (...args: any[]) => ApiExtendsMixin) {
+  class MixedClass extends baseClass implements ApiExtendsMixin {
+    public readonly [_extendsTypes]: HeritageType[] = [];
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public constructor(...args: any[]) {
+      super(...args);
+
+      const options: IApiExtendsMixinOptions = args[0];
+
+      if (this instanceof ApiDeclaredItem) {
+        for (const extendsTokenRange of options.extendsTokenRanges) {
+          this.extendsTypes.push(new HeritageType(this.buildExcerpt(extendsTokenRange)));
+        }
+      } else {
+        throw new InternalError('ApiExtendsMixin expects a base class that inherits from ApiDeclaredItem');
+      }
+    }
+
+    /**
+     * Resolves all of the ApiItem's immediate and inherited members by walking down the
+     * inheritance tree. Be aware that his method returns both immediate and inherited members,
+     * which might not be evident from its name.
```

Comment:
> by walking down the inheritance tree

I've mostly seen class hierarchies portrayed with the base classes (root nodes of the tree) at the top of the diagram. In that representation, the visitor is walking up the tree, not down.